### ISSUE TITLE
Property access does not work on `url.parse(absoluteUrl, true).query` (`URLSearchParams`)

### ISSUE THREAD

**Issue by robhogan** [oven-sh/bun#4902]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

```js
require('url').parse('http://localhost/?foo=bar', true).query.foo
```

### What is the expected behavior?

```js
'bar'
``` 
(the value of `foo` in the query string)

### What do you see instead?

```js
undefined
```
(`query` is a `URLSearchParams` object but direct property access is unavailable)

### Additional information

In Node JS, `query` is a null-prototype object whose entries are query key-values. In Bun, `query` is a `URLSearchParams` object which *in some cases* allows property access consistent with Node JS. In particular this looks to work with relative URLs, but not absolute URLs.

These two URLs have the same query string, but the behaviour in Bun is different:
```js
// bun-url.mjs
import url from 'url';

const pathAndQuery = '/?foo=bar'

const parsedRelative = url.parse(pathAndQuery, true);
console.log('relative', parsedRelative.query, parsedRelative.query.foo);

const parsedAbsolute = url.parse('http://localhost' + pathAndQuery, true);
console.log('absolute', parsedAbsolute.query, parsedAbsolute.query.foo);
```

## In Bun 1.0
```
$ bun bun-url.mjs 
relative URLSearchParams {
  foo: "bar"
} bar 
absolute URLSearchParams {
  "foo": "bar"
} undefined
```

## In Node 20
```
$ node bun-url.mjs
relative [Object: null prototype] { foo: 'bar' } bar
absolute [Object: null prototype] { foo: 'bar' } bar
```

**Comment by Jarred-Sumner** [comment]
Ah yeah this is definitely wrong 

We need to call .toJSON on it

**Comment by panva** [comment]
Don't forget to handle the multiple value case

```
url.parse('?foo=bar&foo=baz', true).query
[Object: null prototype] { foo: [ 'bar', 'baz' ] }
```

**Comment by plutoniumm** [comment]
> Don't forget to handle the multiple value case
> 
> ```
> url.parse('?foo=bar&foo=baz', true).query
> [Object: null prototype] { foo: [ 'bar', 'baz' ] }
> ```

Is this defined in the spec? Because As far as I remeber servers have their own implementations in different situations

some take the last value. Some take the first value some return an array of all values (for repeated key)

**Comment by panva** [comment]
What spec would that be? This is a Node.js API bug.

**Comment by m1212e** [comment]
This bug I have may be related (?):

```
const url = "http://localhost:4000/redirect?session_state=ed781931-95c8-4ffc-9ef9-ce272aca1186&code=f1cec62a-45ea-4949-bdd4-426eed7d55ab.ed781931-95c8-4ffc-9ef9-ce272aca1186.7d8306ef-8267-48b6-84cc-e77889296b14"
const urlObject = new URLSearchParams(url);
urlObject.forEach((value, key) => console.log(`\n${key}: ${value}\n`));
```
outputs:

```
http://localhost:4000/redirect?session_state: ed781931-95c8-4ffc-9ef9-ce272aca1186

code: f1cec62a-45ea-4949-bdd4-426eed7d55ab.ed781931-95c8-4ffc-9ef9-ce272aca1186.7d8306ef-8267-48b6-84cc-e77889296b14
```

So the first parameter has the whole first part of the url as a key, where it only should be the part following the `?`

**Comment by robhogan** [comment]
@m1212e, the [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) constructor takes a query string, not a full URL. That behaviour is consistent with Node and the spec - not a bug.

**Comment by m1212e** [comment]
> @m1212e, the [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) constructor takes a query string, not a full URL. That behaviour is consistent with Node and the spec - not a bug.

Ahhh okay, thanks for the help!

**Comment by hilja** [comment]
I'm trying to run Remix dev server with Bun. Might be related.

```
Could not reach Remix dev server at http://localhost:3001/ping
6393 |        * @param {string} message
6394 |        * @param {string} type
6395 |        */
6396 |       constructor(message, type) {
6397 |         super(message);
6398 |         Error.captureStackTrace(this, this.constructor);
            ^
error: request to http://localhost:3001/ping failed, reason: Can only call URLSearchParams.toJSON on instances of URLSearchParams
 code: "undefined"

      at new FetchBaseError (/Users/bob/web/foo/build/server.js:6398:8)
      at new FetchError (/Users/bob/web/foo/build/server.js:6415:8)
      at /Users/bob/web/foo/build/server.js:7562:17
      at _final (node:http:852:16)
      at callFinal (node:stream:2763:44)
      at prefinish (node:stream:2786:44)
      at finishMaybe (node:stream:2793:48)
      at node:stream:2726:150
      at onend (node:stream:2079:40)
      at endReadableNT (node:stream:2387:52)
      at processTicksAndRejections (:55:76)
```

**Comment by Jarred-Sumner** [comment]
This was fixed in Bun v1.0.1 and we forgot to update this issue

```js
❯ bun --print "require('url').parse('http://localhost/?foo=bar', true).query.foo"
bar


❯ bun-1.1.7 --print "require('url').parse('http://localhost/?foo=bar', true).query.foo"
bar


❯ bun-1.0.1 --print "require('url').parse('http://localhost/?foo=bar', true).query.foo"
bar
```
